### PR TITLE
Allow ENABLE_OPENID_SIGNUP to depend on DISABLE_REGISTRATION

### LINKED ISSUE #1363: Disabling signup in the web installer doesn't disable signups through OpenID
- Gitea version (or commit ref): 1.1.0+66-gc05bd178 (via Docker)
- Git version: 2.8.3
- Operating system: Linux
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [X] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [X] Not relevant
- Log gist:

## Description

I installed Gitea through Docker on my laptop. During the initial configuration, I disabled signup through the web interface. There was nothing about OpenID. As expected, when logged out, there's no "signup" button, only "sign in".

However, I was able to login through OpenID. As the OpenID was unknown, I was redirected to a page where I could choose between connecting this ID to an existing account, or creating a new account. Using this form I was able to create a new account on my local instance, even if signup is disabled.

I checked the config file, and sure enough there was no `[openid]` section.

After creating a new instance and re-running the setup, I edited the config file and added the following:

    [openid]
    ENABLE_OPENID_SIGNUP = true

(as seen [here](https://github.com/go-gitea/gitea/blob/f00a4c8078a74a6cdf64e902b7c8952e3bf4c18c/conf/app.ini#L206))
which fixed the issue.

So:
- either the web installer should include a section about OpenID, and make it possible to disable OpenID and/or signups,
- or the global `DISABLE_REGISTRATION` flag should also disable OpenID signups.

Thanks!

## Screenshots

Disabling signup during setup:

Login with OpenID:

Create new account with OpenID:

Logged in!

### PR DESCRIPTION
**Pull request by strk**
Omit the configuration variable (the default) to be dependent.
Fixes #1363

### CODE REVIEW

**Comment by pgaskin** [comment]
LGTM

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Comment by strk** [comment]
drone reporting broken ? build was successful...

**Comment by strk** [comment]
(comment fixed this - ready to merge)

File: conf/app.ini
Comment (bkcsoft) [inline conf/app.ini:207]: Can we not default this to `%(DISABLE_REGISTRATION)` ?

Comment (bkcsoft) [reply conf/app.ini:207]: That would only work if it's renamed to `DISABLE_OPENID_SIGNUP`, which kinda makes sense anyhow.

**Review by bkcsoft — COMMENTED** [review]
Few thoughts

File: conf/app.ini
Comment (strk) [reply conf/app.ini:207]: Uhm, I didn't know about %(DISABLE_REGISTRATION) - would that use the value of DISABLE_REGISTRATION found under `custom` dir too ? 

Renaming a variable would be a compatibility break, then we'd need to migrate config files ?

Comment (bkcsoft) [reply conf/app.ini:207]: Migrating variablenames could be done by removing the old default from `conf/app.ini` but load it from `custom/conf/app.ini` if it exists. then `cfg = oldVar; if exists(newVar) { cfg = newVar }` in `LoadConfig`. Maybe throw a deprication warning. So no migration needed. And would make the config defautls easier. Since `ENABLE_OPENID_SIGNUP = ! DISABLE_REGISTRATION` which I don't think you can do in `ini` format 😒

Comment (strk) [reply conf/app.ini:207]: Are migration scripts allowed to modify stuff under custom/ ?
In general, is Gitea binary supposed to have write privilege in that dir ?

Comment (strk) [reply conf/app.ini:207]: @bkcsoft I just realized it doesn't take write access to `custom/` to do what you're asking, which is basically still support the old variable in custom if present. Sounds a bit complex though, is that still worth the advantage being the possibility to write `DISABLE_OPENID_REGISTRATION = %DISABLE_REGISTRATION` in `custom/conf/app.ini` ?

**Comment by strk** [comment]
I think this is good enough to be merged. Has 2 LGTM and passed Drone. Fixes #1363 so please merge

**Pull request merged**
